src/main/kotlin/sideChain/iroha/IrohaConsumerImpl.kt

```diff
@@ -0,0 +1,153 @@
+package sideChain.iroha
+
+import notary.IrohaOrderedBatch
+import mu.KLogging
+import notary.Notary
+import main.Configs
+import notary.IrohaCommand
+import java.nio.file.Paths
+import java.math.BigInteger
+
+// iroha
+import ByteVector
+import ModelCrypto
+import ModelTransactionBuilder
+import ModelProtoTransaction
+import Keypair
+import PublicKey
+import kotlin.reflect.jvm.internal.impl.protobuf.InvalidProtocolBufferException
+import iroha.protocol.BlockOuterClass
+import iroha.protocol.CommandServiceGrpc
+
+import io.grpc.ManagedChannelBuilder
+
+/**
+ * Dummy implementation of [IrohaConsumer] with effective dependency
+ */
+class IrohaConsumerImpl(notary: Notary) : IrohaConsumer {
+
+    /** Iroha keypair */
+    val keypair: Keypair
+
+    init {
+        notary.irohaOutput().subscribe {
+            onIrohaEvent(it)
+        }
+        val crypto = ModelCrypto()
+        keypair = crypto.convertFromExisting(
+            readKeyFromFile(Configs.pubkeyPath),
+            readKeyFromFile(Configs.privkeyPath)
+        )
+    }
+
+    /**
+     * Read key from file specified by [path]
+     */
+    private fun readKeyFromFile(path: String): String {
+        try {
+            return String(java.nio.file.Files.readAllBytes(Paths.get(path)))
+        } catch (e: java.io.IOException) {
+            logger.error("Unable to read key files.\n $e")
+            System.exit(1)
+        }
+        return ""
+    }
+
+    /** Convert [ByteVector] to [ByteArray] */
+    private fun toByteArray(blob: ByteVector): ByteArray {
+        val size = blob.size().toInt()
+        val bs = ByteArray(size)
+        for (i in 0 until size) {
+            bs[i] = blob.get(i).toByte()
+        }
+        return bs
+    }
+
+    /** Add [commands] to transaction builder */
+    private fun addCommandToBuilder(
```

Comment:
We have to move convertors from Notary model to Iroha as a dependency and set it into ctor